osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs

```diff
@@ -0,0 +1,340 @@
+    {
+        public const float SELECTION_BORDER_WIDTH = 4;
+
+        private const float transition_duration = 100;
+        private const float content_padding = 10;
+        private const float height = 100;
+        private const float side_strip_width = 5;
+        private const float cover_width = 145;
+
+        private readonly Box selectionBox, sideStrip;
+        private readonly Container coverContainer;
+        private readonly OsuSpriteText name, status, beatmapTitle, beatmapDash, beatmapArtist;
+        private readonly FillFlowContainer<OsuSpriteText> beatmapInfoFlow;
+        private readonly ParticipantInfo participantInfo;
+        private readonly ModeTypeInfo modeTypeInfo;
+
+        private readonly Bindable<string> nameBind = new Bindable<string>();
+        private readonly Bindable<User> hostBind = new Bindable<User>();
+        private readonly Bindable<RoomStatus> statusBind = new Bindable<RoomStatus>();
+        private readonly Bindable<GameType> typeBind = new Bindable<GameType>();
+        private readonly Bindable<BeatmapInfo> beatmapBind = new Bindable<BeatmapInfo>();
+        private readonly Bindable<IEnumerable<User>> participantsBind = new Bindable<IEnumerable<User>>();
+
+        private OsuColour colours;
+        private LocalisationEngine localisation;
+
+        public readonly Room Room;
+
+        public event Action<Visibility> StateChanged;
+
+        private Visibility state = Visibility.Hidden;
+        public Visibility State
+        {
+            get { return state; }
+            set
+            {
+                if (value == state) return;
+                state = value;
+
+                if (state == Visibility.Visible)
+                    selectionBox.FadeIn(transition_duration);
+                else
+                    selectionBox.FadeOut(transition_duration);
+
+                StateChanged?.Invoke(State);
+            }
+        }
+
+        public IEnumerable<string> FilterTerms => new[] { Room.Name.Value };
+
+        private bool matchingFilter;
+        public bool MatchingFilter
+        {
+            get { return matchingFilter; }
+            set
+            {
+                if (value == matchingFilter) return;
+                matchingFilter = value;
+
+                this.FadeTo(MatchingFilter ? 1 : 0, 200);
+            }
+        }
+
+        private Action<DrawableRoom> action;
+        public new Action<DrawableRoom> Action
+        {
+            get { return action; }
+            set
+            {
+                action = value;
+                Enabled.Value = action != null;
+            }
+        }
+
+        public DrawableRoom(Room room)
+        {
+            Room = room;
+
+            RelativeSizeAxes = Axes.X;
+            Height = height + SELECTION_BORDER_WIDTH * 2;
+            Masking = true;
+            CornerRadius = 5 + SELECTION_BORDER_WIDTH / 2;
+
+            Children = new Drawable[]
```

Comment:
this needs to be moved to the `load()` method if possible

Reply:
I think these changes on `DrawableRoom` should be done in a separate PR, as it is not really a part of this one apart from being moved to a new folder. They should probably also be done in `RoomInspector`.